### DISCUSSION TITLE
Check if a variable has some value

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#482] [OPA and Rego] [answered]
I have a quick question.

In Rego language, how to check if a variable has some value and is not undefined?

Thanks.

**Comment by srenatus** [comment]
```rego
default x_is_blah(_) := false
x_is_blah(x) { x == "blah" }
```

Default function values are a new feature, check your OPA version.

**Reply by srenatus** [reply]
Actually this is wrong. I'll get back to you when I'm at a computer, if nobody beats me to it.

**Reply by Ronnie-personal** [reply]
Thanks for the prompt reply. I also need to understand if Rego function supports optional parameter and default return value.

**Reply by srenatus** [reply]
Try object.get

**Comment by Ronnie-personal** [comment]
> ```rego
> default x_is_blah(_) := false
> x_is_blah(x) { x == "blah" }
> ```
> 
> Default function values are a new feature, check your OPA version.

Indeed, it works in OPA 0.55, excellent.

**Comment by Ronnie-personal** [comment]
Thank you for the information, very helpful!
One last question, could you confirm whether Rego function supports optional parameter?

**Reply by srenatus** [reply, marked as answer]
It's not possible -- only via passing objects, as a workaround.
